gc/base/standard/ParallelScavengeTask.cpp

```diff
@@ -88,7 +88,15 @@ MM_ParallelScavengeTask::synchronizeGCThreadsAndReleaseMain(MM_EnvironmentBase *
 	bool result = MM_ParallelTask::synchronizeGCThreadsAndReleaseMain(env, id);
 	uint64_t endTime = omrtime_hires_clock();
 
-	env->_scavengerStats.addToSyncStallTime(startTime, endTime);
+	if (result) {
+		/* Released thread will exit and execute critical section, recored the start time */
+		_syncCriticalSectionStartTime = endTime;
+		_syncCriticalSectionDuration = 0;
+	}
+
+	/* _syncCriticalSectionDuration must be set now, this thread's stall time is at least the duration of critical section */
+	Assert_MM_true((endTime - startTime) >= _syncCriticalSectionDuration);
```

Comment:
Since _syncCriticalSectionStartTime/Duration are introuced by base class (and indeed in theory some other Tasks could benefit from it), It would be nice if we could move all this new logic into the base class method. Then the existing code that sets start/endTime should move too, and it's possible if we hold them in Env rather than on stack. Many other Parallel Tasks could be adjusted to that paradigm, too.

The line below that updates the stats could either stay here, or even be wrapped into a virtual method and go inside base too (in which case, whole this method would not need to be virtualized and specialized for each Parallel Task).

This is ok for now, but something to think about.